pykrige/core.py

```diff
@@ -194,29 +191,70 @@ def _adjust_for_anisotropy(X, center, scaling, angle):
+    y: ndarray
+        float array [n_samples], the input array of values to be kriged
+    variogram_model: str/unicode
+        user-specified variogram model to use
+    variogram_model_parameters: list
+        user-specified parameters for variogram model
+    variogram_function: callable
+        function that will be called to evaluate variogram model
+        (only used if user does not specify variogram model parameters)
+    nlags: int
+        integer scalar, number of bins into which to group inter-point distances
+    weight: bool
+        boolean flag that indicates whether the semivariances at smaller lags
+        should be weighted more heavily in the automatic variogram estimation
+    coordinates_type: str/unicode
+        type of coordinates in X array, can be 'euclidean' for standard rectangular coordinates
+        or 'geographic' if the coordinates are lat/lon
+
+    Returns
+    -------
+    lags: ndarray
+        float array [nlags], distance values for bins into which the semivariances were grouped
+    semivariance: ndarray
+        float array [nlags], averaged semivariance for each bin
+    variogram_model_parameters: list
+        parameters for the variogram model, either returned unaffected if the user specified them or
+        returned from the automatic variogram estimation routine
+    """
 
+    # distance calculation for rectangular coords now leverages scipy.spatial.distance's pdist function,
+    # which gives pairwise distances in a condensed distance vector (distance matrix flattened to a vector)
+    # to calculate semivariances... trick scipy.spatial.distance's pdist function into thinking that the measurements
+    # are in a 2D space by stacking with an array of zeros
     if coordinates_type == 'euclidean':
-        dx = x1 - x2
-        dy = y1 - y2
-        d = np.sqrt(dx**2 + dy**2)
+        d = pdist(X, metric='euclidean')
+        y_temp = np.vstack((y, np.zeros(y.size))).T
+        g = 0.5 * pdist(y_temp, metric='sqeuclidean')
+
+    # geographic coordinates only accepted if the problem is 2D
+    # assume X[:, 0] ('x') => lon, X[:, 1] ('y') => lat
+    # old method of distance calculation is retained here... could be improved in the future
     elif coordinates_type == 'geographic':
-        # Assume x => lon, y => lat
+        if X.shape[1] != 2:
+            raise ValueError('Geographic coordinate type only supported for 2D datasets.')
+        x1, x2 = np.meshgrid(X[:, 0], X[:, 0], sparse=True)
+        y1, y2 = np.meshgrid(X[:, 1], X[:, 1], sparse=True)
+        z1, z2 = np.meshgrid(y, y, sparse=True)
         d = great_circle_distance(x1, y1, x2, y2)
+        g = 0.5 * (z1 - z2)**2.
+        indices = np.indices(d.shape)
+        d = d[(indices[0, :, :] > indices[1, :, :])]
+        g = g[(indices[0, :, :] > indices[1, :, :])]
 
-    g = 0.5 * dz**2
-
-    indices = np.indices(d.shape)
-    d = d[(indices[0, :, :] > indices[1, :, :])]
-    g = g[(indices[0, :, :] > indices[1, :, :])]
```

Comment:
I kept the old style here for the time being because ```pdist``` doesn't natively offer a 'geographic' distance calculation as far as I know. It does accept a user-specified metric function, so eventually with some more cleaning up we can feed the spherical coordinate calculation function to pdist and skip the extra steps of making the full square distance matrix. (Again, just need a vector of unique distances between points for variogram estimation.)